docs/plugins/best_practices.md

```diff
@@ -0,0 +1,175 @@
+# Best practices when developing napari plugins
+
+There are a number of good and bad practices that may not be immediately obvious
+when developing a plugin.  This page
+
+## Do
+
+- **Update your README to make it immediately clear what your plugin does in a
+  few sentences** 
+
+    This is your elevator pitch. You'll loose a lot of potential users quickly
+    if they have to look in the source code, or look through a large
+    documentation page, just to know what the gist of the plugin is.
+
+- **Use images, gifs, or movies to *show* what your plugin does**. 
+
+  Images and movies are *extremely* useful for conveying the usefulness of your
+  plugin. Both github and napari hub make it easy to embed videos in your
+  readme. For example, if you have a dock widget plugin that plots something as
+  the user interacts with the current layer, capture a screengrab and put it in
+  your readme.
+
+  TODO: screengrab tips and code examples. 
+
+
+## Don't
+
+
+- **Don't explicitly declare either `PySide2` or `PyQt5` in your plugin's
+  `install_requires` section.**
+
+    *This is important!*
+
+    Napari supports *both* PyQt and PySide backends for Qt.  It is up to the
+    end-user to choose which one they want. If they installed napari with `pip
+    install napari[all]`, then the `[all]` extra will (currently) install
+    `PyQt5` for them from pypi.  If they installed via `conda install napari`,
+    then they'll have `PyQt5`, but via anaconda cloud instead of pypi. Lastly,
+    they may have installed napari with PySide2
+
+    Here's what can go wrong if you *also* declare one of these backends:
+
+    - If they installed via `conda install napari` and then they install your
+      plugin via `pip` (or via the builtin plugin installer, which currently
+      uses `pip`), then there *will* be a binary incompatibility between the
+      their conda `pyqt` installation, and the new pip "`PyQt5`" installation.
+      *This will very likely lead to a broken environment*. This is an
+      unfortunate consequence of [package naming
+      decisions](https://github.com/ContinuumIO/anaconda-issues/issues/1554),
+      and it's not something napari can fix.
+    - Alternatively, they may end up with *both* PyQt and PySide in their
+      environement, and while that's not always an environment "death sentence",
+      it too can lead to unexpected and difficulty to debug problems.
+
+- **Don't import from PyQt5 or PySide2 in your plugin: use `qtpy`.**
```

Comment:
This seems like something we could enforce with a sanitization tool, ast all modules and look for funky imports (separate issue? 🤔 )

Reply:
yep, could do!